check-configuration.sh

```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
-# This file checks that the settings in resin.env are loaded and authenticating correctly
+# This file checks that the settings in balena.env are loaded and authenticating correctly
 
-source ./resin.env
+source ./balena.env
```

Comment:
If we don't support both file names, then this might be better marked as a major change

Reply:
Oh yes, good catch, 100%

Reply:
What is the preferred method? Should we emit an error if the `resin.env` file exists & exit early (noting the rename), and then only source from `balena.env`? Otherwise I can imagine there being confusion in trying to consolidate two different env files.

Reply:
updated, PTAL

Reply:
Have you tested this? On my machine `source ./non-existent-file.env` prints a 'no such file or directory' error to stderr and exits unsuccessfully. Doesn't look like we're using `set -e` (though arguably we should) so that won't kill the whole script, but it looks like it will still print a messy error message that references resin for every user who doesn't have a legacy config file leftover. 

Imo we should silence that output, and potentially add `set -e` and `|| true` to tidy things up in general.

Reply:
To be clear though: this is a good approach, totally agree that this is the right idea, I'm just quibbling about the details :smiley:

Reply:
Yes I have, the latest commit includes a stub `resin.env` to indicate clearly that the file is required for compatibility, but it's use is discouraged. @pimterry have you checked out the most recent hash, `2f6facdc43a512d6c2b36a770b2b1951`?

Reply:
@xginn8  I don't really like that we have both files (would prefer to check if it exists) but works for me :+1: 
If we used `git mv` then users pulling the new version and re-applying their local changes would end up having their configuration in the brand new `balena.env`.

An easy way to test this is by
* `git stash` your `resin.env`
* create a new branch where you `git mv resin.env balena.env`and commit your changes
* trying to fast-forward your old branch to the new one will warn you to stash, but after the ff poping the stash will result your configuration being applied to `balena.env`

Reply:
Ah, sorry @xginn8, I had assumed that we'd deleted the resin.env file from the repo. I too would prefer to remove resin.env if we can, to avoid the confusion of having both, so that resin.env is only present if there's an old one left around (and as @thgreasi points out, a git mv should handle that for users pulling these scripts using git in an existing directory at least).

I'm going to approve as is, since you're right that this does work, and it's ok, but if people are keen I think we could also potentially improve it further by:

* git moving `resin.env` to `balena.env`, so that users pulling updates with git have their config renamed automatically
* Continuing to `source ./resin.env`, just in case, but silencing errors if it's not present

Reply:
@thgreasi that indeed does sound like a better plan! I had to break the update and move into two separate commits to prevent git from deciding to make a new object. PTAL!